### ISSUE TITLE
[calcite-list]: inconsistent browser focus on <tab>

### ISSUE THREAD

**Issue by jrowlingson** [Esri/calcite-design-system#12881]
**monday.com sync:** #18048608296

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

Initially reported [here](https://github.com/ArcGIS/field-apps-web/issues/4898).

* Tabbing (forward) focuses only the slotted action.
* Shift+Tabbing (backward) focuses the slotted action and the row.

### Actual Behavior

Browser focus traversal via `<tab>` and `<S-tab>` is not symmetric.

### Expected Behavior

Browser focus traversal via `<tab>` and `<S-tab>` is symmetric.

### Reproduction Sample

https://codepen.io/jrowls/pen/OPMLPbw

### Reproduction Steps

* Tabbing (forward) focuses only the slotted action.
* Shift+Tabbing (backward) focuses the slotted action and the row.

### Reproduction Version

3.2.1

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by geospatialem** [comment]
Additional research is needed via a `spike` to determine the `tabindex` support, and if there can be a way to opt out when the value is `0` (e.g., a button or action, or element that is focusable). This was initially offered via an `interactive-container` that previously offered the ability to adjust the tabindex, and should be researched further why it was removed.

**Comment by driskull** [comment]
@jcfranco I would expect our components to honor `tabindex="-1"` set on the host. Is that correct?
